util/dnsname/dnsname.go

```diff
@@ -120,23 +105,22 @@ func isValidFQDN(s string) bool {
 			continue
 		}
 		label := s[st:i]
-		if len(label) == 0 || len(label) > maxLabelLength {
-			return false
-		}
-		if !isalphanum(label[0]) || !isalphanum(label[len(label)-1]) {
+		if !validLabel(label) {
 			return false
 		}
-		for j := 1; j < len(label)-1; j++ {
-			if !isalphanum(label[j]) && label[j] != '-' {
-				return false
-			}
-		}
 		st = i + 1
 	}
 
 	return true
 }
 
+func validLabel(s string) bool {
```

Comment:
probably worth a comment here saying why it's so lax, lest people try to tighten it up again later